elastictransport/elastictransport.go

```diff
@@ -130,6 +131,7 @@ type Client struct {
 	retryBackoff          func(attempt int) time.Duration
 	discoverNodesInterval time.Duration
 	discoverNodesTimer    *time.Timer
+	discoverNodeTimeout   *time.Duration
```

Comment:
I don't think a timeout with a negative value makes sense.
I would change this to not be a pointer, such that we can validate its value is greater than zero.